code/modules/cm_aliens/structures/special/spawn_pool.dm

```diff
@@ -107,11 +108,12 @@
 		return
 
 	for(var/mob/living/carbon/Xenomorph/Larva/L in range(2, src))
-		if(!L.ckey && L.poolable && !QDELETED(L))
-			visible_message(SPAN_XENODANGER("[L] quickly dives into the pool."))
-			linked_hive.stored_larva++
-			linked_hive.hive_ui.update_pooled_larva()
-			qdel(L)
+		if(L.hivenumber == linked_hive.hivenumber)
+			if((!L.ckey || (world.time >= (L.life_time_start + larva_leeway_time))) && L.poolable && !QDELETED(L))
```

Comment:
what actually changes here if ckey is kept

Reply:
Thats original code for ghosted ones, we want ghosted to go away immediately instead of having a delay like the ones with ckeys i've just added.